lib/signs_ui/alerts/alert.ex

```diff
@@ -0,0 +1,44 @@
+defmodule SignsUi.Alerts.Alert do
+  defstruct [:id, :created_at, :service_effect]
+
+  @type id :: String.t()
+
+  @type t :: %__MODULE__{
+          id: id(),
+          created_at: DateTime.t() | nil,
+          service_effect: String.t() | nil
+        }
+
+  defimpl Jason.Encoder, for: SignsUi.Alerts.Alert do
+    def encode(value, opts) do
+      Jason.Encode.map(
+        %{
+          id: value.id,
+          created_at: format_dt(value.created_at),
+          service_effect: value.service_effect
+        },
+        opts
+      )
+    end
+
+    @spec format_dt(DateTime.t() | nil) :: String.t()
+    defp format_dt(nil) do
+      ""
```

Comment:
Can an alert's `created_at` ever be `nil`? I thought this was a non-null field.

Reply:
Well, the API documentation calls it non-null, and same for `service_effect`, but that one, too, I opted for `| nil`. But to me, from the point of view of signs-ui, it's outside its application boundary, so it shouldn't just rely on that assertion. If an alert somehow ends up without a `created_at`, that's not an excuse for signs_ui to go down.

So that implies to me that if we want to internally enforce a model of a required `created_at`, in the later PR when we actually are dealing with alerts, we'd be forced to drop any alerts that are missing that field. But that piece of information didn't seem so critical to any of the functionality that I thought it would warrant that action. If you asked, "hypothetically, if we received an alert without a created_at, would we still want PIOs to be able to select it?" I think the answer is yes, right?

You could make the argument that a missing `created_at` implies data corruption or something, and so we _shouldn't_ show it... but I don't know if that's the case. I actually don't know what the API would do, but I don't at all trust alerts-ui to be robust software that always provides a `created_at` even if it claims to. I suspect the API would still pass that alert along, but I haven't verified that.

Reply:
Hmm, actually no I'm wrong. If the API received an alert missing a `created_at` it [looks like](https://github.com/mbta/api/blob/7dc7446dbcf6e833982b4aba392b2e3aad203b87/apps/parse/lib/parse/alerts.ex#L56) it would 💥 . Given that it hasn't... I guess in this particular way alerts-ui has been pretty consistent at least. I still don't think it's worth signs_ui putting its foot down and demanding its presence here, though.